test/index.js

```diff
@@ -741,6 +741,42 @@ it('accepts request object in validateFunc', (done) => {
     });
 });
 
+it('includes additional attributes in WWW-Authenticate header', (done) => {
+    const server = new Hapi.Server();
+    server.connection();
+    server.register(require('../'), (err) => {
+
+        expect(err).to.not.exist();
+        server.auth.strategy('default', 'basic', 'required', {
+            validateFunc: internals.user,
+            unauthorizedAttributes: { realm: 'hapi' }
+        });
+        server.route({
+            method: 'POST',
+            path: '/',
+            handler: function (request, reply) {
+
+                return reply('ok');
+            },
+            config: {
+                auth: 'default'
+            }
+        });
+
+        const request = { method: 'POST', url: '/', headers: {
+            // intentionally empty
```

Comment:
just to make sure the authorization header is not defined so it definitely goes to the first `Boom.unauthorized`. I can remove it if you prefer